src/components/analyses/landing/AnalysisSubmissionLanding.js

```diff
@@ -132,38 +132,32 @@ export default function AnalysisSubmissionLanding(props) {
     const sharable = analysis ? canShare([analysis]) : false;
     const queryClient = useQueryClient();
 
-    const updateAnalysis = React.useCallback(
-        (notifiMessage) => {
-            const message = notifiMessage?.message;
-            if (message) {
-                const category = message.type;
-                if (
-                    category?.toLowerCase() ===
-                    NotificationCategory.ANALYSIS.toLowerCase()
-                ) {
-                    const analysisId = message.payload?.id;
-                    const status = message.payload?.status;
-                    const resultfolderid =
-                        message.payload?.analysisresultsfolder;
-                    const enddate = message.payload?.enddate;
-
-                    if (analysisId === id && analysis?.status !== status) {
-                        setAnalysis({
-                            ...analysis,
-                            resultfolderid,
-                            status,
-                            enddate,
-                        });
-                    }
-                }
-            }
-        },
-        [analysis, id]
-    );
 
     React.useEffect(() => {
-        updateAnalysis(currentNotification);
-    }, [currentNotification, updateAnalysis]);
+        const message = currentNotification?.message;
+        if (message) {
+            const category = message.type;
+            if (
+                category?.toLowerCase() ===
+                NotificationCategory.ANALYSIS.toLowerCase()
+            ) {
+                const analysisId = message.payload?.id;
+                const status = message.payload?.status;
+                const resultfolderid =
+                    message.payload?.analysisresultsfolder;
+                const enddate = message.payload?.enddate;
+
+                if (analysisId === id && analysis?.status !== status) {
+                    setAnalysis({
+                        ...analysis,
+                        resultfolderid,
+                        status,
+                        enddate,
+                    });
+                }
+            }
+        }
+    }, [analysis, currentNotification, id]);
```

Comment:
Mostly a refactor cos `React.useCallback` or having the logic inside `useEffect` is the same.